Source/Core/DolphinWX/Cheats/GeckoCodeDiag.cpp

```diff
@@ -192,6 +192,9 @@ void CodeConfigPanel::DownloadCodes(wxCommandEvent&)
 
 		while ((std::getline(ss, line).good()))
 		{
+			// Remove \r at the end of the line for files using windows line endings, std::getline only removes \n
+			line = StripSpaces(line);
+
 			// empty line
 			if (0 == line.size() || line == "\r" || line == "\n") // \r\n checks might not be needed
```

Comment:
That sounds like a reasonable explaination. Except that it will only catch empty lines; not lines ending with a \r (as the ```StripSpaces``` does now).